pkg/client/addresses.go

```diff
@@ -320,14 +321,46 @@ func (a *Addresses) BalanceAfterConfirmations(
 	return out, response, nil
 }
 
+type AddressesDataParams struct {
+	matches string
+	keys    []string
+}
+
+type AddressesDataParam func(*AddressesDataParams)
+
+func WithMatches(matches string) AddressesDataParam {
+	return func(p *AddressesDataParams) {
+		p.matches = matches
+	}
+}
+
+func WithKeys(keys []string) AddressesDataParam {
```

Comment:
May be better replace `[]string` with `...string` and later do like this `p.keys = []string(keys)`. The usage will be easier.